c10/core/TensorImpl.h

```diff
@@ -263,6 +277,18 @@ struct C10_API PyInterpreter {
     return (*decref_fn_)(this, pyobj);
   }
 
+  // Perform a detach by deferring to the __torch_dispatch__ implementation of
+  // detach, which will also arrange for the PyObject to get copied in this
+  // situation
+  __ubsan_ignore_function__ c10::intrusive_ptr<TensorImpl> detach(const TensorImpl* self) const {
```

Comment:
In some cases, `shallow_copy_and_detach` is a little more than a detach, it also resets the version counter(?) (see [link](https://github.com/pytorch/pytorch/blob/e7ad82eb2f9856682d95a69114a04655a32c5ad2/torch/csrc/autograd/variable.cpp#L365-L367)). Is this a problem?

Reply:
Oh alright, I guess we need a different operator for this.

Reply:
Yeah, so we could also just manually fix things up after the user implementation returns. TBH, I'm not sure what the right thing to do here.